lib/vanagon/engine/local.rb

```diff
@@ -1,40 +1,37 @@
+require 'vanagon/engine/base'
 require 'vanagon/utilities'
 require 'vanagon/errors'
-require 'benchmark'
 
 class Vanagon
   class Engine
-    class Local
-      attr_accessor :target
+    class Local < Base
 
       def initialize(platform, target = nil)
         @platform = platform
         @target = "local machine"
         @name = 'local'
+        super
+        @required_attributes = Array.new
       end
 
       # Dispatches the command for execution
-      def dispatch(command)
-        puts Benchmark.measure { local_command(command, @workdir) }
-      end
-
-      # Steps needed to tear down or clean up the system after the build is
-      # complete
-      def teardown
-      end
-
-      # This method will take care of validation and target selection all at
-      # once as an easy shorthand to call from the driver
-      def startup(workdir)
-        @workdir = workdir
-        script = @platform.provisioning.join(' && ')
-        dispatch(script)
+      def dispatch(command, return_output = false)
+        if defined?(Bundler)
```

Comment:
Nitpick, but it'd be nice to not repeat the local_command call. You could add a helper like

```
def with_bundler
  if defined?(Bundler)
    Bundler.with_clean_env { yield }
  else
    yield
  end
end
```

Then call it as
```
with_bundler do
  Vanagon::Utilities.local_command(command, return_command_output: return_output)
end
```